dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerOwnershipManager.java

```diff
@@ -149,72 +149,41 @@ public void transferOwnership(
 
   @Override
   @Transactional
-  public void assignOwnership(
-      TrackedEntity trackedEntity,
-      Program program,
-      OrganisationUnit organisationUnit,
-      boolean skipAccessValidation,
-      boolean overwriteIfExists) {
-    if (trackedEntity == null || program == null || organisationUnit == null) {
-      return;
+  public void grantTemporaryOwnership(
+      @Nonnull TrackedEntity trackedEntity, Program program, UserDetails user, String reason)
+      throws ForbiddenException {
+    if (canSkipOwnershipCheck(user, program)) {
+      throw new ForbiddenException(
+          "Temporary ownership not created. Either current user is a superuser, program supplied is does not exist or program supplied is not a tracker program.");
```

Comment:
Here we can easily tell the user exactly what is the issue without the need of reusing `canSkipOwnershipCheck` method.

I think we have enough cases to create a validate method and to give useful error message to the client.

```suggestion
          "Temporary ownership not created. Either current user is a superuser, program supplied does not exist or program supplied is not a tracker program.");
```